### DISCUSSION TITLE
FrSky R9 (not mini, slim, sx, ...)

### DISCUSSION THREAD

**Discussion by xagos** [ExpressLRS/ExpressLRS#2554] [Q&A]
Hey - I have swotched to ELRS today and flashed my R9M2019 sucessfully ;)
Now I am wondering that I am not able to find a bootloader for a (older) R9 receiver from frsky?!
![R9](https://github.com/ExpressLRS/ExpressLRS/assets/13057167/ecca17a1-1475-4866-b851-1c5fd7541c16)
It is based on the STM32F103CBT6 and the SX1276 (and has 8PWM Channels)...

I took a look in the json files (presets) but I did not found anything about, how to setup this receiver.

Perhaps someone can give me additinal information - searching for R9 gives tons of documents to the newer generation fpv mini receivers but nothing about this older models - Thanks in advanced...

**Comment by RA-100** [comment]
I have exactly the same problem. Here are few photos from the inside of the receiver:

![IMG_4618 HEIC_compressed](https://github.com/user-attachments/assets/72b84813-ecd0-4e8a-8425-8d6ec1a846b6)
![IMG_4620 HEIC_compressed](https://github.com/user-attachments/assets/866091fd-739d-42a5-84c1-9ede587e959a)

Any idea, if it is possible to upload ELRS firmware into this receiver?
